docs/vscode.md

```diff
@@ -299,7 +320,9 @@ def foo(x, y, z) {
 To ease the migration of existing scripts, the language server only reports warnings for Groovy-style type annotations and implicit variable declarations. These warnings will become errors in the future.
 
 :::{note}
-Type annotations and static type checking will be addressed in a future version of the Nextflow language specification.
+Because type annotations are useful in practice for providing type checking at runtime, the language server will not report errors or warnings for Groovy-style type annotations, so as to not burden existing scripts that make heavy use of them.
+
+Type annotations will be addressed in a future version of the Nextflow language specification, at which point the language server will provide a way to automatically migrate Groovy-style type annotations to the new syntax.
```

Comment:
```suggestion
Type annotations are useful for providing type checking at runtime. To avoid burdening existing scripts that use them heavily, the language server will not report errors or warnings for Groovy-style type annotations.

Type annotations will be addressed in a future version of the Nextflow language specification. At this point, the language server will provide a way to automatically migrate Groovy-style type annotations to the new syntax.
```